src/caffe/layers/cudnn_conv_layer.cpp

```diff
@@ -12,6 +12,66 @@
 
 namespace caffe {
 
+cudnnConvolutionFwdAlgo_t
+GetCuDNNFwdAlgo(ConvolutionParameter_CuDNNFwdAlgorithm algo) {
+    switch (algo) {
+      case ConvolutionParameter_CuDNNFwdAlgorithm_IMPLICIT_GEMM:
+          return CUDNN_CONVOLUTION_FWD_ALGO_IMPLICIT_GEMM;
+      case ConvolutionParameter_CuDNNFwdAlgorithm_IMPLICIT_PRECOMP_GEMM:
+          return CUDNN_CONVOLUTION_FWD_ALGO_IMPLICIT_PRECOMP_GEMM;
+      case ConvolutionParameter_CuDNNFwdAlgorithm_GEMM:
+          return CUDNN_CONVOLUTION_FWD_ALGO_GEMM;
+      case ConvolutionParameter_CuDNNFwdAlgorithm_DIRECT:
+          return CUDNN_CONVOLUTION_FWD_ALGO_DIRECT;
+      case ConvolutionParameter_CuDNNFwdAlgorithm_FFT:
+          return CUDNN_CONVOLUTION_FWD_ALGO_FFT;
+      case ConvolutionParameter_CuDNNFwdAlgorithm_FFT_TILING:
+          return CUDNN_CONVOLUTION_FWD_ALGO_FFT_TILING;
+#if CUDNN_VERSION_MIN(5, 0, 0)
+      case ConvolutionParameter_CuDNNFwdAlgorithm_WINOGRAD:
+          return CUDNN_CONVOLUTION_FWD_ALGO_WINOGRAD;
+#endif
+      default:
+          return CUDNN_CONVOLUTION_FWD_ALGO_IMPLICIT_GEMM;
```

Comment:
This mean that if you ask for Winograd with cuDNN v4, you will silently fallback to IMPLICIT_GEMM instead. I'm not sure that's a behavior we want. Maybe we should assert, or at least log it?